internal/database/repository/queue_int_test.go

```diff
@@ -495,6 +495,203 @@ func TestAddQueues_Integration(t *testing.T) {
+			},
+			queueToUpdate: &dao.PartitionQueueDAOInfo{
+				Partition:       "default",
+				QueueName:       "root",
+				CurrentPriority: 0,
+				Children: []dao.PartitionQueueDAOInfo{
+					{
+						Partition:       "default",
+						QueueName:       "root.org",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+					{
+						Partition:       "default",
+						QueueName:       "root.system",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Update when new child queues has been added",
+			existingQueues: []*dao.PartitionQueueDAOInfo{
+				{
+					Partition:       "default",
+					QueueName:       "root",
+					CurrentPriority: 0,
+				},
+			},
+			queueToUpdate: &dao.PartitionQueueDAOInfo{
+				Partition:       "default",
+				QueueName:       "root",
+				CurrentPriority: 0,
+				Children: []dao.PartitionQueueDAOInfo{
+					{
+						Partition:       "default",
+						QueueName:       "root.org",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+					{
+						Partition:       "default",
+						QueueName:       "root.system",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Update when both parent queue changed and new child queues has been added",
+			existingQueues: []*dao.PartitionQueueDAOInfo{
+				{
+					Partition:       "default",
+					QueueName:       "root",
+					CurrentPriority: 0,
+				},
+			},
+			queueToUpdate: &dao.PartitionQueueDAOInfo{
+				Partition:       "default",
+				QueueName:       "root",
+				CurrentPriority: 100,
+				Children: []dao.PartitionQueueDAOInfo{
+					{
+						Partition:       "default",
+						QueueName:       "root.org",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+					{
+						Partition:       "default",
+						QueueName:       "root.system",
+						Parent:          "root",
+						IsLeaf:          true,
+						CurrentPriority: 200,
+					},
+				},
+			},
+			expectedError: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// clean up the table after the test
+			t.Cleanup(func() {
+				_, err := connPool.Exec(ctx, "DELETE FROM queues")
+				if err != nil {
+					t.Fatalf("could not empty queue table: %v", err)
+				}
+			})
+			// seed the existing queues
+			if tt.existingQueues != nil {
+				if err := repo.AddQueues(ctx, nil, tt.existingQueues); err != nil {
+					t.Fatalf("could not seed queue: %v", err)
+				}
+			}
+			// update the new queue
+			err := repo.UpdateQueue(ctx, tt.queueToUpdate)
+			if tt.expectedError != nil {
+				require.Error(t, err)
+				// match expected error message exist in the actual error message
+				assert.Contains(t, err.Error(), tt.expectedError.Error())
```

Comment:
You can leverage early return here so it is easier to read:
```go
if tt.expectedErr {
  // do the logic
  return
}
// rest of the assertions in case error is not expected
```